components/common/hooks/useCWSAvailabilityCheck.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+
+import {Client4} from 'mattermost-redux/client';
+
+export default function useCWSAvailabilityCheck() {
+    const [canReachCWS, setCanReachCWS] = useState(false);
+    useEffect(() => {
+        Client4.cwsAvailabilityCheck().then(() => setCanReachCWS(true));
+    });
```

Comment:
I think we want to add an empty dependency array to the `useEffect`. Otherwise, each component render will re-run the effect, calling the endpoint lots of times.

Reply:
Ah, good catch! Thanks :)